server/src/main/java/org/elasticsearch/reservedstate/service/ReservedClusterStateService.java

```diff
@@ -186,8 +188,7 @@ static boolean isNewError(ReservedStateMetadata existingMetadata, Long newStateV
             || existingMetadata.errorMetadata().version() < newStateVersion);
     }
 
-    private void saveErrorState(ErrorState errorState) {
-        ClusterState clusterState = clusterService.state();
+    private void saveErrorState(ClusterState clusterState, ErrorState errorState) {
```

Comment:
This change to pass in the most recent cluster state from the cluster state update task is sort of tangential, it just ensures we hit the issue more reliably. ClusterService.state() might not be the latest state, but the one we get in the update task will be.